test/CadlRanchProjectsNonAzure.Tests/type-property-nullable.cs

```diff
@@ -0,0 +1,248 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Xml;
+using Scm._Type.Property.Nullable;
+using Scm._Type.Property.Nullable.Models;
+using AutoRest.TestServer.Tests.Infrastructure;
+using Azure.Core;
+using NUnit.Framework;
+using System.ClientModel;
+
+namespace CadlRanchProjectsNonAzure.Tests
+{
+    public class TypePropertyNullableTests : CadlRanchNonAzureTestBase
+    {
+        [Test]
+        public Task Type_Property_Nullable_String_getNonNull() => Test(async (host) =>
+        {
+            var response = await new NullableClient(host, null).GetStringClient().GetNonNullAsync();
+            Assert.AreEqual("foo", response.Value.RequiredProperty);
+            Assert.AreEqual("hello", response.Value.NullableProperty);
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_String_getNull() => Test(async (host) =>
+        {
+            var response = await new NullableClient(host, null).GetStringClient().GetNullAsync();
+            Assert.AreEqual("foo", response.Value.RequiredProperty);
+            Assert.AreEqual(null, response.Value.NullableProperty);
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_String_patchNonNull() => Test(async (host) =>
+        {
+            var value = new
+            {
+                requiredProperty = "foo",
+                nullableProperty = "hello"
+            };
+            var response = await new NullableClient(host, null).GetStringClient().PatchNonNullAsync(BinaryContent.Create(BinaryData.FromObjectAsJson(value)));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_Nullable_String_patchNull() => Test(async (host) =>
+        {
+            string value = "{ \"requiredProperty\": \"foo\", \"nullableProperty\": null }";
+            var response = await new NullableClient(host, null).GetStringClient().PatchNullAsync(BinaryContent.Create(BinaryData.FromString(value)));
```

Comment:
I do not see a point why we have to change to a different way to build the body.
Could we change back to the way we have in above test case?